src/alire/alire_early_elaboration.adb

```diff
@@ -2,12 +2,46 @@ with Alire;
 --  Kind of circularity here but somehow it slips past static GNAT rules??
 
 with GNAT.Command_Line;
+with GNAT.IO;
 with GNAT.OS_Lib;
 
-with Simple_Logging;
+with Simple_Logging.Filtering;
 
 package body Alire_Early_Elaboration is
 
+   ----------------
+   -- Add_Scopes --
+   ----------------
+
+   procedure Add_Scopes (Debug_Arg : String) is
+      --  Receives as-is the --debug/-d[ARG] argument. This is a list of
+      --  optionally comma-separated, plus/minus prefixed substrings that will
+      --  be used for filtering against the enclosing entity/source location.
+      --  Example whitelisting argument: +commands,-search
+      --  Example blacklisting argument: -commands,+search
+      --  The first sign puts the filter in (-) blacklist / (+) whitelist mode.
+      --  In whitelist mode, only the given substrings are logged, unless later
+      --  added as exception. E.g., in the "+commands,-search" example, only
+      --  commands traces would be logged (because of whitelist mode), except
+      --  the ones for the search command (because given as an exception).
+      --  In the "-commands,+search" example for blacklist mode, everything but
+      --  command traces would be logged, but search command traces would be
+      --  logged because that's the exception.
+
+      --  Once scopes are used, we activate logging of enclosing entity and
+      --  location to provide full logging information.
+
+   begin
+      if not Simple_Logging.Filtering.Add_From_String (Debug_Arg,
+                                                       Say => True)
+      then
+         --  Bypass debug channel, which was not entirely set up.
+         --  Otherwise we get unwanted location/entity info already.
+         GNAT.IO.Put_Line ("ERROR: Invalid logging filters.");
```

Comment:
Why using GNAT.IO instead of Ada.Text_IO?

Reply:
I guess at some point that code was in a preelaborable library. I'll change it to plain Ada.Text_IO.